Mixpanel/MixpanelInstance.swift

```diff
@@ -845,19 +850,42 @@ extension MixpanelInstance {
      - parameter completion: an optional completion handler for when the flush has completed.
      */
     open func flush(completion: (() -> Void)? = nil) {
-        serialQueue.async() {
+        
+        self.archive()
+
+        networkQueue.async() {
```

Comment:
i agree. this was the reason why we did serial queue async -> network queue async. it makes it so that if users do track and then flush it guarantees that track is in the flush, like they expect